carbonmark-api/src/routes/projects/get.utils.ts

```diff
@@ -62,39 +59,81 @@ export const getDefaultQueryArgs = async (
  * Chooses between default and selective price.
  * Sorted cheapest first.
  */
-export const getOffsetTokenPrices = (
-  offset: GetCarbonOffsetsByProjectAndVintageQuery["carbonOffsets"][number],
+type ProjectBalances = Record<
+  string,
+  { balance: string; projectAddress: string }
+>;
+
+export const getDigitalCarbonTokenPrices = (
+  digitalCarbonProject: CarbonProjectType,
   poolPrices: Record<string, PoolPrice>
 ) => {
+  const poolAddressToKey: Record<string, string> = {};
+
+  for (const [key, poolInfo] of Object.entries(POOL_INFO)) {
+    poolAddressToKey[poolInfo.poolAddress.toLowerCase()] = key;
+  }
+
+  const tokenBalances: ProjectBalances = {
+    nbo: { balance: "0", projectAddress: "" },
+    ubo: { balance: "0", projectAddress: "" },
+    bct: { balance: "0", projectAddress: "" },
+    nct: { balance: "0", projectAddress: "" },
+    mco2: { balance: "0", projectAddress: "" },
+  };
+
+  /**
+   * @todo mc02 poolPrices are NaN because there is no default project
+   */
+
+  const credits = digitalCarbonProject.carbonCredits;
+
+  for (const credit of credits) {
+    if (credit.poolBalances.length === 0) continue;
+
+    for (const poolBalance of credit.poolBalances) {
+      const poolId = poolBalance.pool.id;
+      const key = poolAddressToKey[poolId];
+
+      tokenBalances[key] = {
+        balance: poolBalance.balance,
+        projectAddress: credit.id, // should this be this be credit.id or poolBalance.pool.id ?
```

Comment:
I'm not 100% here but I think it's the `credit.id` as you have already put in here..

@Atmosfearful of @cujowolf can you confirm?

Reply:
Yep this should be the project credit address. This field will need to be updated once we incorporate the ICR tons into the subgraph as the `credit.id` will no longer be guaranteed to be an ethereum address.